src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -217,24 +184,14 @@ object RuleUtils {
         val relation = HadoopFsRelation(
           newLocation,
           new StructType(),
-          readSchema,
+          StructType(index.schema.filter(baseRelation.schema.contains(_))),
           if (useBucketUnion) Some(index.bucketSpec) else None,
           new ParquetFileFormat,
           Map())(spark)
 
-        if (filesDeleted.isEmpty) {
-          val updatedOutput =
-            baseOutput.filter(attr => relation.schema.fieldNames.contains(attr.name))
-          baseRelation.copy(relation = relation, output = updatedOutput)
-        } else {
-          val lAttr = AttributeReference(IndexConstants.DATA_FILE_NAME_COLUMN, StringType)(
-            NamedExpression.newExprId)
-          val updatedOutput =
-            baseOutput.filter(attr => relation.schema.fieldNames.contains(attr.name))
-          val deletedFileNames = filesDeleted.map(Literal(_)).toArray
-          val rel = baseRelation.copy(relation = relation, output = updatedOutput ++ Seq(lAttr))
-          Project(updatedOutput, Filter(Not(In(lAttr, deletedFileNames)), rel))
```

Comment:
@pirz You can refer this commit (of - delta lines) for delete support.

Reply:
Btw, since @pirz still has some time before he works on this, it might be better if he somehow depends on this method *after* your delete PR gets in (it also makes sense since this is already in `RuleUtils.scala`). What do you think?